### PR TITLE
varnish-cache-5.2 compatibility

### PR DESCRIPTION
**Pull request by dmatetelki**
- detecting varnish-cache version at least 5.2 with `pkg-config`
- setting up the variables to match the varnish-cache-5.2 - `LIBVARNISHAPI_2.0`
- the rest of the behaviour is the same as with `HAVE_VARNISH_V4`

### CODE REVIEW

**Comment by mfournier** [comment]
Awesome @dmatetelki, thanks so much for working on this !

It seems the build fails with this error, when building against varnish 5.0:

```
src/varnish.c: In function 'varnish_read':
src/varnish.c:1393:5: error: implicit declaration of function 'VSM_Destroy' [-Werror=implicit-function-declaration]
     VSM_Destroy(&vd);
     ^
cc1: all warnings being treated as errors
Makefile:6384: recipe for target 'src/varnish_la-varnish.lo' failed
make[1]: *** [src/varnish_la-varnish.lo] Error 1
```

Should this one be `VSM_Delete()` instead, as we're in a `#if HAVE_VARNISH_V3 || HAVE_VARNISH_V4` block here ?

Also, would you be so kind and make this pull-request against the collectd-5.8 branch instead of master, so we can benefit from it in the upcoming release due end of November ?

Thanks a lot !

File: src/varnish.c [hunk 1]
```diff
@@ -152,7 +152,17 @@ static int varnish_monitor(void *priv,
...
-#if HAVE_VARNISH_V4
+#if HAVE_VARNISH_V5
+  char namebuff[100];
+  char *c;
+
+  c = rindex(pt->name, '.');
+  strcpy(namebuff, c + 1);
+  name = namebuff;
+
+  (void)class;
```
Comment (octo) [inline src/varnish.c]: `class` seems to only be *read* in the `HAVE_VARNISH_V3` block. Can we remove this from all the other implementations, including this one to simplify the code?

**Comment by mfournier** [comment]
@dmatetelki, I just pushed 2 commits on top of yours, hopefully addressing the comments above.

Curiously the CI build doesn't seem to have been triggered. I'll go ahead and rebase these 4 patches on top of the collectd-5.8 branch and duplicate this PR, to force the CI checks. Sorry for the mess...

**Comment by dmatetelki** [comment]
Thank you, @mfournier, for fixing my mistake!

@octo: Can you close this PR once it's merged, or shall I do it?

**Comment by octo** [comment]
Closing this in favor of #2527

**Pull request closed without merging**